<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
    "http://www.w3.org/TR/html4/strict.dtd">
<html lang="en">
      
<head>
	<title>StatAlign &ndash; Home</title>
	<link rel="stylesheet" type="text/css" href="StatAlign.css"></link>
	<script>
	  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');	  
	  ga('create', 'UA-59661515-1', 'auto');
	  ga('send', 'pageview');	  
</script>
</head>
<body>
<div class="allContent">
	<div class="headerImage">
	<div class="pad">
		<table class="logo">
		<tr>
		<td><img class="logo" alt="StatAlign" src="StatAlignLogo.png" width="300"></td>
		<td class="subtitle">
			<h3>An Extendable Software Package for Joint Bayesian Estimation of Alignments and Evolutionary Trees</h3>
		</td>
		</tr>
		</table>
	</div>
	</div>

	<div id="menu-wrap">
	<div id="menu">
		<ul>
			<li class="first"><a href="index.html">Home</a></li>
			<li><a href="downloads.html">Downloads</a></li>
			<li><a href="doc/user_manual.html">User's manual</a></li>
			<li><a href="doc/developer_manual.html">For developers</a></li>
			<li><a href="references.html">References</a></li>
		</ul>
	</div>
	</div>

	<div class="pad">
	<div class="container">
		<h3>What is StatAlign?</h3>
		<i>StatAlign</i> is an extendable software package for Bayesian analysis of protein, DNA and RNA sequences. Multiple alignments, phylogenetic trees and evolutionary parameters	are co-estimated in a Markov Chain Monte Carlo framework, allowing for reliable quantification of the uncertainty in these estimates.
		<p>
		This approach accounts for the inherent interdependence between estimation of alignments and phylogenies, avoiding the inherent bias present in traditional methods that rely on a single alignment. This helps to avoid artifacts such as those resulting from the choice of guide tree when constructing alignments.
		<p>
		The models behind the analysis permit the comparison of evolutionarily distant sequences. The TKF92 insertion-deletion model allows for indel information to be used to help inform phylogeny estimation, and can be coupled with a wide variety of substitution models. A broad range of models for nucleotide and amino acid data is included in the package and the plugin management system ensures that new models can be easily added.
		<p>
		StatAlign also includes model extension plugins capable of incorporating protein and RNA structural information, thereby increasing the reliability of the inference.
		<p>
		  Although the increased computational overheads of this approach limit its application to smaller datasets (we have tested up to 20-30 sequences), the probabilistic nature of the model allows for more reliable information to be obtained from smaller datasets, and enables the uncertainty in the results to be quantified more accurately. We are currently working on a parallelised version that will extend the capabilities of StatAlign to much larger datasets.
		 <p>

		<p>
		<p>StatAlign is available from the <a href="downloads.html">Downloads</a> page.<br><br>
		For more information, please see our papers on the <a href="references.html">References</a> page.
		<br><br>

		<h3>Recent news</h3>
		<font size="-1"><i>10 Feb 2015</i></font><br>
		<p>
		StatAlign v3.2 is released (<a href="https://raw.github.com/statalign/statalign/master/CHANGELOG.txt" target="_blank">What's new in this version?</a>)
		<p>
		Joint sequence-structure analysis is now possible for cases where only some sequences have known structure, expanding the range of datasets that can be analysed using the StructAlign plugin.
		<p>
		<font size="-1"><i>4 Jun 2014</i></font><br>
		<p>
		Paper describing StructAlign published in <i>Molecular Biology and Evolution:</i><br><br>
		Herman JL, Challis CJ, Nov&aacute;k &Aacute;, Hein J and Schmidler, SC (2014) Simultaneous Bayesian estimation of alignment and phylogeny under a joint model of protein sequence and structure. <i>Molecular Biology and Evolution</i>, <b>31</b>(9):2251-2266. <a href="http://www.ncbi.nlm.nih.gov/pubmed/24899668" target="_blank">PubMed</a> <a href="http://mbe.oxfordjournals.org/content/31/9/2251" target="_blank">MBE website</a>
		<p>
		<font size="-1"><i>24 Nov 2013</i></font><br>
		<p>
		StatAlign v3.1 is out! (<a href="https://raw.github.com/statalign/statalign/master/CHANGELOG.txt" target="_blank">What's new in this version?</a>)
		<p>
		Proposal variance for substitution parameter is now tuned automatically. The StructAlign plugin can now read and write from PDB files, and the maximum likelihood structural superposition is written to a PDB file. Structural B-factors and pairwise RMSDs are used to annotate the current alignment sample in GUI mode.
		<p>
		<font size="-1"><i>26 Oct 2013</i></font><br>
		<p>
		StatAlign v3.0 released! (<a href="https://raw.github.com/statalign/statalign/master/CHANGELOG.txt" target="_blank">What's new in this version?</a>)
		<p>
		New features include:
		<ul>
		<li>Option to keep tree fixed when phylogeny is known</li>
		<li>StructAlign plugin for alignment of protein structures</li>
    		<li>Introduced framework for handling extensions to the evolutionary model</li>
    		<li>Significant improvements to mixing on tree topologies</li>
    		<li>Several new types of MCMC moves to improve mixing and convergence</li>
    		<li>Automatic tuning for MCMC moves</li>
    		<li>Restructured code to allow for easy addition of new MCMC move objects</li>
    		<li>Code now allows for prior distributions to be easily specified</li>
    		<li>Selection of standard priors added (Gaussian, Gamma, etc.)</li>
		</ul>
		<p>
		<font size="-1"><i>24 Feb 2013</i></font><br>
		<p>
		StatAlign v2.1 has been released! (<a href="https://raw.github.com/statalign/statalign/master/CHANGELOG.txt" target="_blank">What's new in this version?</a>)
		<p>
		Since version 2.0 StatAlign has built-in RNA-specific features, which include RNA
		secondary structure prediction from multiple alignments using either a thermodynamic
		approach (RNAalifold) or a Stochastic Context-Free Grammar approach (PPfold).
		This methodology allows more reliable structure predictions by incorporating
		alignment uncertainty. See our paper on the <a href="references.html">References</a>
		page and on <a href="http://www.ncbi.nlm.nih.gov/pubmed/23335014"
		target="_blank">PubMed</a>.
		
		<br><br>
	</div>
	</div>
</div>
</body>
</html>
