<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
    "http://www.w3.org/TR/html4/strict.dtd">
<html lang="en">
      
<head>
	<title>StatAlign &ndash; Home</title>
	<link rel="stylesheet" type="text/css" href="StatAlign.css"></link>
</head>
<body>
<div class="allContent">
	<div class="headerImage">
	<div class="pad">
		<table class="logo">
		<tr>
		<td><img class="logo" alt="StatAlign" src="StatAlignLogo.png" width="300"></td>
		<td class="subtitle">
			<h3>An Extendable Software Package for Joint Bayesian Estimation of Alignments and Evolutionary Trees</h3>
		</td>
		</tr>
		</table>
	</div>
	</div>

	<div id="menu-wrap">
	<div id="menu">
		<ul>
			<li class="first"><a href="index.html">Home</a></li>
			<li><a href="downloads.html">Downloads</a></li>
			<li><a href="doc/user_manual.html">User's manual</a></li>
			<li><a href="doc/developer_manual.html">For developers</a></li>
			<li><a href="references.html">References</a></li>
		</ul>
	</div>
	</div>

	<div class="pad">
	<div class="container">
		<h3>Recent news</h3>
		<font size="-1"><i>26 Oct 2013</i></font><br>
		<p>
		StatAlign v3.0 released! (<a href="https://raw.github.com/statalign/statalign/master/CHANGELOG.txt" target="_blank">What's new in this version?</a>)
		<p>
		New features include:
		<ul>
		<li>Option to keep tree fixed when phylogeny is known</li>
		<li>StructAlign plugin for alignment of protein structures</li>
    		<li>Introduced framework for handling extensions to the evolutionary model</li>
    		<li>Significant improvements to mixing on tree topologies</li>
    		<li>Several new types of MCMC moves to improve mixing and convergence</li>
    		<li>Automatic tuning for MCMC moves</li>
    		<li>Restructured code to allow for easy addition of new MCMC move objects</li>
    		<li>Code now allows for prior distributions to be easily specified</li>
    		<li>Selection of standard priors added (Gaussian, Gamma, etc.)</li>
		</ul>
		<p>StatAlign is available from the <a href="downloads.html">Downloads</a> page.<br><br>
		<p>
		<font size="-1"><i>24 Feb 2013</i></font><br>
		<p>
		StatAlign v2.1 has been released! (<a href="https://raw.github.com/statalign/statalign/master/CHANGELOG.txt" target="_blank">What's new in this version?</a>)
		<p>
		Since version 2.0 StatAlign has built-in RNA-specific features, which include RNA
		secondary structure prediction from multiple alignments using either a thermodynamic
		approach (RNAalifold) or a Stochastic Context-Free Grammar approach (PPfold).
		This methodology allows more reliable structure predictions by incorporating
		alignment uncertainty. See our paper on the <a href="references.html">References</a>
		page and on <a href="http://www.ncbi.nlm.nih.gov/pubmed/23335014"
		target="_blank">PubMed</a>.
		
		<br><br>
		<h4>What is StatAlign?</h4>
		<i>StatAlign</i> is an extendable software package for Bayesian
		analysis of Protein, DNA and RNA sequences.
		Multiple alignments, phylogenetic trees and evolutionary parameters
		are co-estimated in a Markov Chain Monte Carlo framework, allowing
		for reliable measurement of the accuracy of the results.
		<p>
		This approach eliminates common artifacts that traditional methods
		suffer from, at the cost of increased computational time. These
		artifacts include the dependency of the constructed phylogeny on a
		single (probably suboptimal) alignment and bias towards the guide
		tree upon which the alignment relies.
		<p>
		The models behind the analysis permit the comparison of evolutionarily
		distant sequences: the TKF92 insertion-deletion model can be coupled
		to an arbitrary substitution model. A broad range of models for
		nucleotide and amino acid data is included in the package and the
		plug-in management system ensures that new models can be easily added.
		<p>
		For more information, please see our papers on the <a href="references.html">References</a> page.
		<br><br>
	</div>
	</div>
</div>
</body>
</html>
