<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
    "http://www.w3.org/TR/html4/strict.dtd">
<html lang="en">
      
<head>
	<title>StatAlign &ndash; User's manual</title>

<script>
if(document.URL.indexOf("http:") == 0) {
document.write(' \
<link rel="stylesheet" type="text/css" href="../StatAlign.css"></link> \
');}
</script>

</head>
<body>

<script>
if(document.URL.indexOf("http:") == 0) {
document.write(' \
<div class="allContent"> \
	<div class="headerImage"> \
	<div class="pad"> \
		<table class="logo"> \
		<tr> \
		<td><img class="logo" alt="StatAlign" src="../StatAlignLogo.png" width="300"></td> \
		<td class="subtitle"> \
			<h3>An Extendable Software Package for Joint Bayesian Estimation of Alignments and Evolutionary Trees</h3> \
		</td> \
		</tr> \
		</table> \
	</div> \
	</div> \
 \
	<div id="menu-wrap"> \
	<div id="menu"> \
		<ul> \
			<li><a href="../index.html">Home</a></li> \
			<li><a href="../downloads.html">Downloads</a></li> \
			<li class="first"><a href="user_manual.html">User\'s manual</a></li> \
			<li><a href="developer_manual.html">For developers</a></li> \
			<li><a href="../references.html">References</a></li> \
		</ul> \
	</div> \
	</div> \
 \
	<div class="pad"> \
	<div class="container"> \
');} else {
document.write('<div style="padding: 0 5px 0 5px">');
}
</script>

<!-- CONTENT_START -->

<h2>StatAlign user's manual</h2>
<h4>For StatAlign v3.1 &ndash; last updated 24 Nov 2013</h4>

<i>The most up-to-date version of this document is available from
<a href="http://statalign.github.com/doc/user_manual.html">StatAlign's website</a>.</i>

<a name="contents">
<h2>Contents</h2>
	[ <a href="#introduction">Introduction</a> ]
		<div style='padding-left: 2em'>
		[ <a href="#about_help">About this manual</a> ]<br>
		[ <a href="#about_statalign">About StatAlign</a> ]<br>
		</div>
	[ <a href="#menubar">The menu bar</a> ]<br>
		<div style='padding-left: 2em'>
		[ <a href="#file">File</a> ]<br>
		[ <a href="#analysis">Analysis</a> ]<br>
		[ <a href="#model">Model</a> ]<br>
		[ <a href="#help">Help</a> ]<br>
		</div>
	[ <a href="#settings">Analysis settings</a> ]
		<div style='padding-left: 2em'>
		[ <a href="#io_parameters">Output preferences</a> ]<br>
			<!--
			<div style='padding-left: 2em'>
			[ <a href="#logfile">Logfile</a> ]<br>
			[ <a href="#postprocess_files">Postprocess files</a> ]<br>
			[ <a href="#alignment_formats">Alignment formats</a> ]<br>
			</div>
			-->
		[ <a href="#mcmc_params">MCMC parameters</a> ]<br>
		</div>
	[ <a href="#postprocesses">The graphical interface panels</a> ]<br>
		<div style='padding-left: 2em'>
		[ <a href="#sequences">Sequences</a> ]<br>
		[ <a href="#alignment">Alignment</a> ]<br>
		[ <a href="#loglikelihood">LogLikelihood</a> ]<br>
		[ <a href="#mpd">MPD</a> ]<br>
		[ <a href="#tree">Tree</a> ]<br>
		[ <a href="#consensus_tree">Consensus tree</a> ]<br>
		[ <a href="#consensus_network">Consensus network</a> ]<br>
		</div>	
	<!--[ <a href="#example">An example</a> ]<br>-->
	[ <a href="#structalign_plugin">The protein structure (StructAlign) plugin</a> ]<br>
		<div style='padding-left: 2em'>
		[ <a href="#structalign_introduction">Introduction</a> ]<br>	
		[ <a href="#structalign_data">Loading structural data</a> ]<br>		
		[ <a href="#structalign_params">Visualising parameter convergence</a> ]<br>		
		[ <a href="#structalign_ali">Structural annotations on alignment</a> ]<br>		
		[ <a href="#structalign_output">Output generated</a> ]<br>		
		[ <a href="#structalign_refs">References</a> ]<br>		
		</div>
	[ <a href="#rna_plugin">The RNA plugin</a> ]<br>
		<div style='padding-left: 2em'>
		[ <a href="#rna_tutorial">Tutorial</a> ]<br>
		[ <a href="#rna_introduction">Introduction</a> ]<br>
		[ <a href="#activating_rna_mode">Activating the RNA mode</a> ]<br>
		[ <a href="#rna_mode_settings">RNA mode settings</a> ]<br>
		[ <a href="#rna_visualisation">RNA mode visualisation</a> ]<br>
		[ <a href="#entropy">Information entropy</a> ]<br>
		[ <a href="#similarity">Similarity</a> ]<br>
		[ <a href="#rna_output">RNA mode file output</a> ]<br>
		</div>	

<!--------------------------------------------------------------------------------------------------->
<a name="introduction">
<h2>Introduction</h2>
	[ <a href="#tutorial">Tutorial</a> ]
	[ <a href="#about_help">About this manual</a> ]
	[ <a href="#about_statalign">About StatAlign</a> ]		

	<a name="about_help">
	<h3>About this manual</h3>
	
		<p align="justify">
		This is a comprehensive manual for the graphical interface to StatAlign. 
		Command-line-specific information can be obtained by running
		<pre>
		  java -jar StatAlign.jar -help
		</pre>

		The documentation on this page can be navigated by using the hyperlinks and 
		the <tt>Forward</tt> and <tt>Backward</tt> buttons of your browser. 
		This manual can also be accessed from within StatAlign, via the 
		<tt>Help</tt> &#8594; <tt>User's manual</tt> menu.
		When using the latter method, clicking the <tt>Home</tt> button reloads this page.<br><br>
		
		The topics are arranged in hierarchial order, the <img src="img/up.jpg" width="10" height="10" border=0>
		arrow moves up one level.<br><br>
		
		The <img src="img/out.jpg" height=15 width=15 border=0> arrow indicates a hyperlink
		outside of the program. You cannot open such a link if your computer
		is not connected to the internet. Since these hyperlinks point to third party
		pages, we cannot guarantee the content of the linked pages. If you discover a broken link,
		please let us know.
		<br><br>
		</p>
		
		[ <a href="#introduction">Introduction</a> ] 
		<a href="#introduction"><img src="img/up.jpg" width="10" height="10" border=0></a>

	<a name="about_statalign">
	<h3>About StatAlign</h3>

		<p align="justify">
		StatAlign provides a unified approach for inferring multiple
		sequence alignments, evolutionary trees and model parameters within a joint Bayesian framework.<br><br>

		Most other methods for phylogenetic inference use a single fixed alignment as input. However, this
		can result in significant bias in the resulting trees, since the results may be very sensitive to
		the specific choice of alignment. In addition, these methods typically treat 
		insertions and deletions (gaps) as missing data, discarding a great deal of important information
		in the process, and potentially further biasing the results.<br><br>
		
		To address these problems, StatAlign jointly samples multiple alignments, evolutionary trees
		and model parameters under a stochastic model of substitution, insertion and deletion
		<a href="http://www.biomedcentral.com/1471-2105/9/137" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>, making use of a Markov chain Monte Carlo scheme
		<a href="http://en.wikipedia.org/wiki/Markov_chain_Monte_Carlo" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>
		to generate samples from the desired posterior distribution.
		The program allows a range of different substitution models; the insertion-deletion model 
		is a modification of the TKF92 model
		<a href="http://www.ncbi.nlm.nih.gov/pubmed/1556741?dopt=Abstract" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>, which allows 'refragmentation' at internal nodes. For more details,
		see Mikl&oacute;s <i>et al.</i> (2008)
		<a href="http://www.biomedcentral.com/1471-2105/9/137" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>.
		
		<br><br>
		
		Users can load sequences, and choose a substitution model with which to analyze
		the sequences. Once the parameters of the Markov chain and desired output are specified
		(e.g. log-likelihood trace, alignment, tree samples etc.), the MCMC chain can be initialised,
		generating a set of samples for the parameters of interest, as well as 
		consensus alignments and trees.
		
		The following sections contain descriptions of the graphical elements of the program (menu bar, 
		tabulated panels, pop-up dialog windows), with examples illustrating how an analysis can be carried
		out.<br><br>
		</p>
		
		[ <a href="#introduction">Introduction</a> ] 
		<a href="#introduction"><img src="img/up.jpg" width="10" height="10" border=0></a>
	
	
[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
<!--------------------------------------------------------------------------------------------------->
<a name="menubar">
<h2>The menu bar</h2>

	[ <a href="#file">File</a> ]
	[ <a href="#analysis">Analysis</a> ]
	[ <a href="#model">Model</a> ]
	[ <a href="#help">Help</a> ]

	<a name="file">
	<h3>File</h3>
		
		<p align="justify">
		In the File menu, we can load sequences (and other types of data if supported by plugins), 
		set the output preferences and exit the program.
		
		<h4><u>Add sequence(s)...</u></h4>
		<p align="justify">
		Clicking this menu item will result the pop up of a file opening dialog window.
		We can browse the directories in this window and open a file. The file to be opened 
		should contain sequences in Fasta file 
		<a href="http://en.wikipedia.org/wiki/Fasta_format" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0></a>
		format (and potentially other types of data, when appropriate plugins exist). 
		Sequences can be viewed and removed via the 
		<a href="#sequences"><tt>Sequences</tt></a> panel.</p>
		
		<h4><u>Output settings...</u></h4>
		<p align="justify">
		Clicking this menu item will pop up a dialog window where
		you can set up your
		<a href="#io_parameters">output preferences</a></p>
		
		<h4><u>Exit</u></h4>
		<p align="justify">
		Exits StatAlign.</p>
		
		[ <a href="#menu">The menu bar</a> ] 
		<a href="#menu"><img src="img/up.jpg" width="10" height="10" border=0></a>
		<!---------------------------------------------------------------------------------->		
	<a name="analysis">
	<h3>Analysis</h3>
		
		<h4><u>Run</u></h4>
		<p align="justify">
		After loading your sequences and choosing your <a href="#model">Model</a> and <a href="#io_parameters">Output settings</a>, you can click this menu item to set the MCMC parameters and start the MCMC analysis.

		<p>Information will be given about the progress of the MCMC sampling in the Status
		bar, such as the number of burn-in steps completed and the number of MCMC samples
		taken. The <a href="#postprocesses">Postprocessing panels</a> will show both snapshots
		from the MCMC chain, as well as various summaries calculated from the MCMC samples.

		<!--<p>When the MCMC analysis terminates you will be notified with a pop-up window that
		lists all the output files created.-->

		<h4><u>Pause</u></h4>
		<p align="justify">
		We can pause an active MCMC run by clicking this button.
		</p>	

		<h4><u>Resume</u></h4>
		<p align="justify">
		A suspended run can be continued by clicking here.
		</p>	
		
		<h4><u>Stop</u></h4>
		<p align="justify">
		This item stops an MCMC run. Output files will still be created from the samples
		that have been taken so far, but you won't be able to restart the process from the
		same point. This should be used only if you indeed want to terminate the MCMC run early.
		</p>	
				
		[ <a href="#menu">The menu bar</a> ] 
		<a href="#menu"><img src="img/up.jpg" width="10" height="10" border=0></a>
		<!---------------------------------------------------------------------------------->
		
		
		<!---------------------------------------------------------------------------------->		
		
	<a name="model">
	<h3>Model</h3>

		<p align="justify">
		This menu is for selecting the substitution model that is used for analyzing the sequences.
		<i>StatAlign</i> automatically recognises the sequences, and if sequences cannot be
		analysed by the chosen model (for example, protein sequences are forced to be
		analysed by a nucleotide model), a pop-up error box notifies the user.<br>
		<!--<center><img src="img/bad_model.png" width=400 height=256></center><br>-->
		<p align="justify">
		Implemented models are:
		
		<ul>
		<li>Amino acid substitution models
			<ul> 
				<li>Blosum rate
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/8234244" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Dayhoff
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/15483331" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Chloroplast
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/10795826" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Jones-Taylor-Thornton
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/1633570" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Mammalian mitochondrial
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/9732458" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Mitochondrial reversible
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/8642615" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Retrovirus model
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/12165843" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>M&uuml;ller-Vingron model
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/11382360" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>WAG, Whelan &amp; Goldman model
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/11319253" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
			</ul>
		</li>
		<li>Nucleic acid substitution models
			<ul>		
				<li>Felsenstein81
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/7288891" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Felsenstein84<br>
					Felsenstein, J. 1984. Distance methods for inferring phylogenies: a justification. Evolution 38:16Ð24.
					</li>
					
				<li>Hasegawa-Kishino-Yano  (HKY85)
					<a href="http://www.cs.manchester.ac.uk/ai/Software/phase/manual/node90.html" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Jukes-Cantor 
					<a href="http://www.geocities.com/royrajan/Web_Project/methods.html#JukesCantor" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
			
				<li>Kimura 3 parameter
					<a href="http://www.pnas.org/cgi/content/abstract/78/1/454" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>

				<li>General reversible
					<a href="http://www.cs.manchester.ac.uk/ai/Software/phase/manual/node92.html" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Tamura 1992
					<a href="http://www.cs.manchester.ac.uk/ai/Software/phase/manual/node93.html" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Tamura-Nei
					<a href="http://www.cs.manchester.ac.uk/ai/Software/phase/manual/node91.html" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>			
			</ul>
			</li>
		</ul>
		</p>
		
		[ <a href="#menu">The menu bar</a> ] 
		<a href="#menu"><img src="img/up.jpg" width="10" height="10" border=0></a>
		<!---------------------------------------------------------------------------------->		
	<a name="help">
	<h3>Help</h3>
		The Help menu provides access to help files and other useful information in a browser
		or Java's mini-browser.
		<ul>
			<li>About...<br>
				A short summary of the authors and used libraries.</li>
			
			<li><a href="user_manual.html">User's manual</a><br>
				This document.</li>
			
			<li><a href="developer_manual.html">Developer's manual</a><br>
				A detailed description of how to implement novel plugins for StatAlign.</li>
		
			<li><a href="javadoc/index.html" target="_blank">Javadocs for StatAlign</a><br>
				Javadoc generated descriptions of the public classes, methods
				and variables of StatAlign.</li>
			
		</ul>
		
		[ <a href="#menu">The menu bar</a> ] 
		<a href="#menu"><img src="img/up.jpg" width="10" height="10" border=0></a>
		<!---------------------------------------------------------------------------------->		
	

[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
<!--------------------------------------------------------------------------------------------------->

<a name="settings">
<h2>Analysis settings</h2>

	[ <a href="#io_parameters">Output preferences</a> ]
	[ <a href="#mcmc_params">MCMC parameters</a> ]

	<a name="io_parameters">
	<h3>Output preferences</h3>

		[ <a href="#logfile">Logfile</a> ]
		[ <a href="#postprocess_files">Postprocess files</a> ]
		[ <a href="#alignment_formats">Alignment formats</a> ]

		<p align="justify">
		The output preferences are available via <tt>File</tt> &#8594; <tt>Output settings...</tt>.
		It will open this pop-up window:
		<center><img src="img/output_preferences.png" width=400 height=256></center><br>
		<p align="justify">	
		Below we describe the possible settings in this window.
		</p>

		<a name="logfile">
		<h4>Logfile</h4>
		
			<p align="justify">
			On the top left corner of the output preferences window, users can choose which
			postprocesses write into the logfile. The format of the output
			is the following:</p>
			<pre>
		Sample [sample_number] [TAB] postprocess_name: [TAB] data
			</pre>
			<p align="justify">
			For example:</p>
			<pre>
		Sample 0        Loglikelihood:  -3718.11540541795
		Sample 0        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.04547):0.01,(P1_1ala:0.02321,P1_1avha:0.01238):0.07713):0.01):0.03021,P1_2ran:0.08613);
		Sample 1        Loglikelihood:  -3620.5414903951587
		Sample 1        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.04547):0.01,(P1_1ala:0.02321,P1_1avha:0.07235):0.07713):0.01):0.03021,P1_2ran:0.12924);
		Sample 2        Loglikelihood:  -3618.189925496592
		Sample 2        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.04547):0.01,(P1_1ala:0.02321,P1_1avha:0.11236):0.07713):0.01):0.04352,P1_2ran:0.11424);
		Sample 3        Loglikelihood:  -3615.7331381880526
		Sample 3        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.04547):0.01,(P1_1ala:0.02321,P1_1avha:0.11236):0.07713):0.01837):0.04352,P1_2ran:0.11424);
		Sample 4        Loglikelihood:  -3587.894867267956
		Sample 4        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.06347):0.01,(P1_1ala:0.02321,P1_1avha:0.11236):0.07713):0.01837):0.04352,P1_2ran:0.11424);
		Sample 5        Loglikelihood:  -3585.9665576563943
		Sample 5        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.06347):0.01,(P1_1ala:0.02321,P1_1avha:0.11236):0.08542):0.01837):0.04352,P1_2ran:0.11902);
		Sample 6        Loglikelihood:  -3551.4723341911595
		Sample 6        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.06347):0.05301,(P1_1ala:0.02321,P1_1avha:0.11236):0.08542):0.03414):0.04352,P1_2ran:0.11902);
			.
			.
			.
			</pre>	
			</p>

		[ <a href="#io_parameters">Output preferences</a> ]
		<a href="#io_parameters"><img src="img/up.jpg" width="10" height="10" border=0></a>

		<a name="postprocess_files">
		<h4>Postprocess files</h4>

			<p align="justify">
			Users also can choose which postprocesses generate their own output file. Note
			that this file has its own format, and might be different from the data written into the log file.
			For example, the tree postprocess writes the sampled trees into a standard Nexus file
			<a href="http://www.bioperl.org/wiki/Nexus_tree_format" target="_blank">
			<img src="img/out.jpg" height=15 width=15 border=0>
			</a>
			as its own output file.
			</p>
		[ <a href="#io_parameters">Output preferences</a> ]
		<a href="#io_parameters"><img src="img/up.jpg" width="10" height="10" border=0></a>

		<a name="alignment_formats">
		<h4>Alignment formats</h4>
		<p align="justify">
		Supported alignment formats are:
		<ul>
			<li><a name="statalign_format">StatAlign<br>
				<p align="justify">	
				It prints both the sequences at the leaves and the sequences
				predicted for the internal nodes. The sequence names and
				the alignment is separated by a TAB. The predicted sequences for the
				internal nodes contains and empty record for their name.
				Each sequence in the alignment is shown in one line. 
			</li>
			
			<li>Clustal<br>
				<p align="justify">
				The first line contains the word "Clustal".
				The alignment is broken into 60 character wide blocks.
				The sequence names are shown at the begining of the line.
				Alignment columns with all matches are marked with an
				asteriks. See also 
				<a href="http://www.bioperl.org/wiki/ClustalW_multiple_alignment_format" target="_blank">
				<img src="img/out.jpg" height=15 width=15 border=0>
				</a>		
			</li>
			
			<li>Fasta<br>
				<p align="justify">
				Alignment names are marked with a "&gt;" symbol
				and are in a separated line. Alignments are broken into
				60 character wide blocks. Each sequence are printed separately, so
				it is not easy to see which character is aligned to which one.
				See also
				<a href="http://www.bioperl.org/wiki/FASTA_multiple_alignment_format" target="_blank">
				<img src="img/out.jpg" height=15 width=15 border=0>
				</a>		
			</li>
			
			<li>Phylip<br>
				<p align="justify">
				The first line contains the number of sequences and the
				length of the alignment. The alignment is broken into 60
				character wide blocks, and a space is inserted after each 10 character long
				part. The name of the sequences are shown at the begining of the
				first block. See also
				<a href="http://www.bioperl.org/wiki/PHYLIP_multiple_alignment_format" target="_blank">
				<img src="img/out.jpg" height=15 width=15 border=0>
				</a>
			</li>
			
			<li>Nexus<br>
				<p align="justify">
				Nexus has its own <tt>XML</tt>-like format. A nexus alignment contains
				a header containing information about the seuqence type, sequence names, etc.
				See also
				<a href="http://bio.perl.org/wiki/NEXUS_multiple_alignment_format" target="_blank">
				<img src="img/out.jpg" height=15 width=15 border=0>
				</a>
			</li>
		</ul>
		</p>
		[ <a href="#io_parameters">Output preferences</a> ]
		<a href="#io_parameters"><img src="img/up.jpg" width="10" height="10" border=0></a>

	[ <a href="#settings">Analysis settings</a> ]
	<a href="#settings"><img src="img/up.jpg" width="10" height="10" border=0></a>

	<a name="mcmc_params">
	<h3>Setting the MCMC parameters</h3>

		<p align="justify">
		The MCMC parameters dialog window can now be opened by clicking on the tool icon 
		<img src="img/settings.png" width="20" height="20" border=0>, which opens up the MCMC settings dialogue panel.<br><br>
		<center><img src="img/mcmc_settings.png" width=400 height=256></center><br>
		In this panel, the following parameters can be set before launching the MCMC run:
		<ul>
			<li>Burn-in cycles.<br>
			<p align="justify">
			The Markov chain underlying the MCMC simulation takes a certain amount of time to 
			converge to its equilibrium distribution. Only when this convergence is complete
			will the samples be generated from the correct posterior distribution that we are interested in. 
			This convergence period is called burn-in phase. Users can set
			how many steps at the beginning of the Markov chain should be treated as
			the burn-in period. The number of steps necessary for convergence may vary significantly
			for different datasets, and it is important to check that the chain has successfully converged.
			</li>
			
			<li>Cycles after burn-in<br>
			<p align="justify">
			This is the number of Markov chain steps to be taken
			after the burn-in phase. During these steps, the samples are assumed to be from the 
			desired posterior distribution, and are used for inference.
			</li>
			
			<li>Sampling rate, <i>c</i><br>
			<p align="justify">
			After each <i>c</i> steps during the 
			post burnin-in period, the state of the Markov chain is recorded, 
			and 
			<a href="#postprocesses">postprocessed</a>. 
			</li>
			
			<li>Seed<br>
			<p align="justify">
			The Java language provides a random number generator that generates the
			pseudo-random numbers
			<a href="http://en.wikipedia.org/wiki/Pseudorandom_number_generator" target="_blank">
			<img src="img/out.jpg" height=15 width=15 border=0>
			</a>
			based on a so-called <i>seed</i>. Users can set this seed here.
			The program generates the same result when the same seed is used. 
			This is useful when one wants to reproduce the result of a run. 
			In order to check convergence of the Markov chain, the analysis
			should be repeated with the same dataset using different random
			seeds, to generate multiple independent runs. 			
			</li>

			<li>Randomisation period<br>
			<p align="justify">
			This specifies the length of period at the start of the MCMC simulation during 
			which the state of the chain should be randomly perturbed by accepting all 
			proposed moves. The purpose of this is to test the sensitivity of the final 
			results to the initial conditions. The main randomisation occuring in this period
			is for the tree topology, since most other parameters typically lose memory of their
			initial state very quickly. A randomisation period of a few hundred steps combined
			with different random seeds is typically sufficient to ensure that multiple MCMC
			runs start from significantly different configurations. 
			</li>
		</ul>
		
		<br><br>
<!--
		Alternatively, the user can choose to automate the process of setting one or more of the parameters by checking the corresponding checkboxes:
		<br><br>
		<center><img src="img/parameters1.png" border=0></center><br>
		Automation is not enabled by default because it may result in the termination of the MCMC process before convergence. Our recommendation is that you use the automatic settings to get a reasonable sampling setup when you don't have the resources to experiment with different MCMC settings.
		
		</p>
-->		
	[ <a href="#settings">Analysis settings</a> ]
	<a href="#settings"><img src="img/up.jpg" width="10" height="10" border=0></a>

[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
<!--------------------------------------------------------------------------------------------------->

<a name="postprocesses">
<h2>The graphical interface panels</h2>
	[ <a href="#sequences">Sequences</a> ]
	[ <a href="#alignment">Alignment</a> ]
	[ <a href="#loglikelihood">LogLikelihood</a> ]
	[ <a href="#mpd">MPD</a> ]
	[ <a href="#tree">Tree</a> ]
	[ <a href="#consensus_tree">Consensus tree</a> ]
	[ <a href="#consensus_network">Consensus network</a> ]

	<a name="sequences">
	<h3>Sequences (and other data)</h3>

		<p align="justify">
		Sequences can be loaded from the menu, following 
		<tt>File</tt> &#8594; <tt>Add sequence(s)...</tt>, or by selecting the
		link in the welcome page.
		<center><img src="img/load_sequences.png" width=400 height=256></center><br>
		The loaded sequences are shown on the Sequences tabulated panel,
		in standard Fasta format. For easier visualisation, the name of the
		sequences are highlighted with dark blue, and printed with different
		type of fonts.<br><br>
		<center><img src="img/input_sequences.png" width=400 height=256></center><br>
		<p align="justify">
		To remove a sequence from the list, click into the desired sequence and
		then to the <tt>Remove</tt> button. Adding sequences can be done simply
		by following <tt>File</tt> &#8594; <tt>Add sequence(s)...</tt>.<br>

		The sequences are automatically recognised if they are nucleotide or
		protein sequences, and a default substitution model is associated to them.
		If the sequences are not recognised, an error message is shown, indicating
		what are the unknown characters in the input sequences.<br><br>

		When other types of data are added and associated with each sequence, for example
		<a href="#structalign_introduction">protein structures</a>, these will also appear 
		alongside the sequence name.
		</p>

		
	
		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>


	<a name="alignment">
	<h3>Alignment</h3>
		
		<p align="justify">
		The MCMC algorithm takes a random walk on the joint distribution of alignments, trees and
		evolutionary parameters. The current alignment can be seen on the <tt>Alignment</tt>
		panel. Above the alignment is shown the posterior probability of each alignment column,
		as estimated from the samples taken so far. This is shown in blue once the burn-in 
		period is over.</p>
		<center><img src="img/alignment.png" width=400 height=256></center><br>

		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>

	<a name="loglikelihood">
	<h3>LogLikelihood</h3>

		<p align="justify">
		The log-likelihood trace is printed onto the screen, to assist with assessing 
		convergence of the chain. The burn-in phase is coloured in red, and the post burnin-in
		phase in blue:
		<center><img src="img/loglikelihood.png" width=400 height=256></center><br>
		Note that convergence of the log-likelihood is not sufficient
		to judge convergence of the entire MCMC chain, but lack of convergence of the 
		log-likelihood certainly indicates that the chain has not converged. 
		Once the log-likelihood has converged, the distributions of other parameters can be
		checked to determine whether convergence has been achieved. 
		</p>

		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>

	<a name="mpd">
	<h3>Maximum Posterior Decoding alignment</h3>
	
		<p align="justify">
		The program estimates summarises the alignment samples so far to produce a 
		type of consensus alignment, using <i>maximum posterior decoding</i> (MPD)
		<a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2324098" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>. 
		The posterior probabilities for each column are printed on top of the alignment, indicating
		the reliability of the corresponding MPD alignment column.
		<center><img src="img/mpd.png" width=400 height=256></center><br>
		<p align="justify">
		The MPD alignment is estimated using only alignment samples from the after-burn-in
		period, hence the alignment is not shown during the burn-in phase in this panel.
		</p>

		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>		
		
	<a name="tree">
	<h3>Tree</h3>
	
		<p align="justify">
		This panel shows the current tree in the Markov chain.
		<center><img src="img/tree.png" width=400 height=256></center><br>
		</p>			

		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>
	<a name="consensus_tree">
	<h3>Consensus tree</h3>
	
		<p align="justify">
		This panel shows the majority consensus tree constructed from all the tree samples taken so far
		<center><img src="img/consensus_tree.png" width=400 height=256></center><br>
		</p>			

		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>
	<a name="consensus_network">
	<h3>Consensus network</h3>
	
		<p align="justify">
		This is similar to the consensus tree, except that unresolved splits are shown as 
		cycles in a graph, indicating all sampled relationships. 
		<center><img src="img/consensus_network.png" width=400 height=256></center><br>
		</p>			

		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>



[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
<!--------------------------------------------------------------------------------------------------->
<!--
<a name="example">
<h2>An example</h2>
	
	<p align="justify">
	In this example, we will analyze pfcp (Papain Family Cystein Proteinase) 
	sequences. First we open the file containing 13 sequences downloaded from the
	HOMSTRAD database
	<a href="http://tardis.nibio.go.jp/homstrad/" target="_blank">
	<img src="img/out.jpg" height=15 width=15 border=0>
	</a>:
	<center><img src="img/example1.png" width=400 height=256>
	<img src="img/example2.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	We can see the loaded sequences in the <tt>Sequences</tt> panel:
	<center><img src="img/example3.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">	
	Then we set the output preferences. We select all possible posterior analyzis and
	output files, and we opt to get alignments in nexus format:
	<center><img src="img/example4.png" width=400 height=256>
	<img src="img/example5.png" width=400 height=256>
	<img src="img/example6.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	After this we start the MCMC analysis with the default parameters (10000 burn-in
	steps, 100000 steps after burn-in, sampled after each 1000 steps, random seed is set to
	1):
	<center><img src="img/example7.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	We have to give the name of the output files, and after this the analysis starts:
	<center><img src="img/example7a.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">	
	The first state is generated by an iterative alignment, which takes time. Indeed,
	this step takes as much time as a Clustal-W multiple alignment itself. <i>StatAlign</i>
	warns the users that the initialization takes time:
	<center><img src="img/example8.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">		
	We can see the loglikelihood values quickly increasing in the burn-in phase:
	<center><img src="img/example9.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	The loglikelihood trace analysis reveals that the 10000 steps were enough for convergence, 
	however, there is a significant autocorrelation, and hence 100000 steps were not enough to
	get a sufficient sample size:
	<center><img src="img/example10.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	Indeed, these sequences are very hard to analyze, as they contain several variable region.
	Small posterior probabilities on top of the MPD alignment indicate these regions:
	<center><img src="img/example11.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	The sampled trees are written in nexus format, and can be opened with SplitsTree4. 
	The consensus network analysis of SplitsTree4 shows the phylogenetic uncertainty:
	<center><img src="img/example12.png" width=400 height=256>
	</center><br>
	</p>

	[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
--->
<br><br>

<a name="structalign_plugin">
<h2>The protein structure (StructAlign) plugin</h2>
		[ <a href="#structalign_introduction">Introduction</a> ]<br>	
		[ <a href="#structalign_data">Loading structural data</a> ]<br>		
		[ <a href="#structalign_params">Visualising parameter convergence</a> ]<br>		
		[ <a href="#structalign_ali">Structural annotations on alignment</a> ]<br>		
		[ <a href="#structalign_output">Output generated</a> ]<br>		
		[ <a href="#structalign_refs">References</a> ]<br>		

<a name="structalign_introduction">
<h3>Introduction</h3>
The protein structure plugin, StructAlign, extends the evolutionary model of StatAlign, allowing for protein structures to be used 
to help with estimating alignments and phylogenies. Since protein structures typically diverge much more slowly than sequences, 
this enables evolutionary relationships to be more accurately and reliably recovered even for distantly related proteins. 
StructAlign also estimates branch-specific rates of structural evolution, which can be used to help infer episodes of 
increased or decreased selective pressure on structures. (See <a href="#structalign_refs">References</a> for more details.)


<br><br>
Here we describe the StructAlign options available through the graphical interface.
Additional options may be available via the command line interface, and can be listed by running
<pre>
java -jar StatAlign.jar -help:structal
</pre>

<a name="structalign_data">
<h3>Loading structural data</h3>
The current version of the StructAlign plugin requires that each sequence have an associated 
structure before the MCMC run can be started. 
(In future versions it will also be possible to run with structures associated with
only some of the sequences.)<br><br>

Protein structures can be loaded and associated with sequences in one of two ways:
<ol> 
<li> reading in sequences and structures directly from a <tt>.pdb</tt> file</li>
<li> separately or simultaneously loading sequences, and structures from a <tt>.coor</tt> file</li>
</ol>

The first method is the simplest way to read in sequences and structures together, and simply requires 
choosing <tt>File</tt> &#8594; <tt>Add sequence(s)...</tt>, and selecting a valid PDB
<a href="http://www.pdb.org" target="_blank">
<img src="img/out.jpg" height=15 width=15 border=0> 
</a>
file from the filesystem. This file should contain only a single chain. If multiple conformations are present for any atoms
then the first conformation will be used. The alpha carbon atom is used as the location for each residue.
Several structures can be added simultaneously using this
method, by selecting multiple PDB files at the same time, by clicking and using the <tt>Ctrl</tt> key 
(or the <tt>Command</tt> key on a Mac).<br><br>

<center><a href="img/structalign_screenshot_1.png"><img src="img/structalign_screenshot_1.png" width=400></a></center>
<br><br>
The second method requires that sequence files <a href="#sequences">already be loaded</a>, or selected and read
in at the same time as the structures. Although this method requires more preliminary steps than the 
PDB-based method, it allows for additional flexibility for users with advanced requirements.
The coordinates of the protein structures should be contained in a <tt>.coor</tt> file
with the following format:
<pre>
>name1
36.587	49.012	31.672	 54.01
35.023	49.494	28.184	 53.01
...     ...     ...      ...
>name2
58.559	46.653	71.532	 62.34
57.828	45.597	67.975	 55.06
...     ...     ...      ...
</pre>
The first three columns represent the x, y and z coordinates of a single atom associated with each
residue in the structure. Typically this will be the alpha carbon. The name of the structure
should appear before the start of the coordinates, and the names (<tt>name1</tt> and <tt>name2</tt> in the 
above example) should match the names of the sequences to which the structures will be associated. 
The fourth column is optional, and contains the crystallographic B-factors associated with each
triplet of atomic coordinates. If present, the B-factor data is used to allow structural heterogeneity
to be incorporated into the evolutionary model. (This information is included by default when 
structures are read in from a PDB file.)
<br><br>

<center><a href="img/structalign_screenshot_1a.png"><img src="img/structalign_screenshot_1a.png" width=400></a></center>
<br><br>
Whichever way structures are read in, summary information will be displayed alongside the associated sequence in the Sequence panel.
When each sequence has an associated structure, the MCMC run can be started.<br><br>

<center><a href="img/structalign_screenshot_2.png"><img src="img/structalign_screenshot_2.png" width=400></a></center>
[ <a href="#structalign_plugin">The protein structure (StructAlign) plugin</a> ] 
<a href="#structalign_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="structalign_params">
<h3>Visualising parameter convergence</h3>
To assist with monitoring convergence of the parameters associated with the structural model, there
is a GUI panel that shows trace plots for each parameter. Values in the burn-in period are shown in red,
and after the burn-in, in blue. When all of these trace plots have stabilised, this is
a good indicator that the MCMC chain has converged to the desired stationary distribution. Also shown in the
top-left of these plots is the current acceptance ratio for MCMC moves on each parameter. Ideally these
should end up between 0.2 and 0.4. If any of these values are significantly outside this range once the
burn-in is complete, it may imply that a longer <a href="#mcmc_params">burn-in period</a> should be used.<br><br>

<center><a href="img/structalign_screenshot_3.png"><img src="img/structalign_screenshot_3.png" width=400></a></center>
<a name="structalign_ali">

<h3>Structural annotations on alignment</h3>
As an additional aid to analysing the output, two additional tracks are added above the current alignment. 
These correspond to the average pairwise RMSD for each column (red), and the predicted heterogeneity
derived from crystallographic B-factors when available (green). These are shown alongside the posterior probability
of the alignment column (blue). When the RMSD postprocessing plugin is enabled, the pairwise RMSD and B-factor
associated with each column in the maximum likelihood alignment are output to a <tt>.rmsd.mle</tt> file.<br><br>

<center><a href="img/structalign_screenshot_4.png"><img src="img/structalign_screenshot_4.png" width=400></a></center>
[ <a href="#structalign_plugin">The protein structure (StructAlign) plugin</a> ] 
<a href="#structalign_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="structalign_output">
<h3>Output generated</h3>
For each run with the StructAlign plugin enabled, a set of additional output files is generated.
These files all contain as the base name a prefix corresponding to the name of the
first data file read in, denoted by <tt>FILE</tt> below.
<ul>
<li><tt>FILE.struct</tt> &ndash; structural model parameters for each MCMC sample</li>
<li><tt>FILE.super.pdb</tt> &ndash; maximum likelihood superposition in PDB format</li>
<li><tt>FILE.rmsd</tt> (optional) &ndash; average pairwise RMSD and tree distances between each leaf</li>
<li><tt>FILE.rmsd.mle</tt> (optional) &ndash; average pairwise RMSD and B-factors for each column in the maximum likelihood alignment</li>
</ul>
The <tt>.struct</tt> file contains posterior samples for each of the unknown parameters that are estimated as part of the StructAlign model:<br><br>

<table border="1">
<tr><td>Parameter name</td><td>Meaning</td></tr>
<tr><td>&tau;</td><td>Overall structural variance (similar to squared radius of gyration, units of &#8491;<sup>2</sup>)</td></tr>
<tr><td>&epsilon;</td><td>Amount of structural variability attributed to background (non-evolutionary) fluctuations</td></tr>
<tr><td>&sigma;<sup>2</sup><sub>g</sub></td><td>Global structural diffusivity (&#8491;<sup>2</sup> / subst. per site)</td></tr>
<tr><td>&sigma;<sup>2</sup><sub>k</sub></td><td>Branch-specific structural diffusivity (&#8491;<sup>2</sup> / subst. per site)</td></tr>
<tr><td>&nu;</td><td>Variance of branch-specific diffusivity parameters (on a log scale)</td></tr>
</table><br>
The first line of the <tt>.struct</tt> file contains the names of the parameters, and each column contains a set of posterior
samples for the specified parameter. Each row corresponds to one MCMC iteration, beginning after the burn-in period has ended.
<br><br>
The RMSD trace can be switched on via the <a href="#io_parameters">Output settings</a> menu. 
The maximum likelihood superposition is outputted to a PDB file, where each structure is present as a separate chain,
beginning at 'A'. This can be read by most structural viewing programs, for example VMD 
<a href="http://www.ks.uiuc.edu/Research/vmd/" target="_blank">
<img src="img/out.jpg" height=15 width=15 border=0> 
</a>. <br><br>
<center><a href="img/superposition.png"><img src="img/superposition.png" width=300></a></center>
<br><br>
[ <a href="#structalign_plugin">The protein structure (StructAlign) plugin</a> ] 
<a href="#structalign_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>
[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="structalign_refs">
<h3>References</h3>
JL Herman, CJ Challis, &Aacute; Nov&aacute;k, J Hein and SC Schmidler, "Simultaneous Bayesian estimation of alignment and phylogeny under a joint sequence-structure model" (<i>submitted, November 2013</i>)<br>
JL Herman, &Aacute; Nov&aacute;k, CJ Challis, SC Schmidler and J Hein, "StatAlign 3: statistical alignment and phylogenetics with protein structures'' (<i>submitted, November 2013</i>)

<br><br>
[ <a href="#structalign_plugin">The protein structure (StructAlign) plugin</a> ] 
<a href="#structalign_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>
[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>


<a name="rna_plugin">
<h2>The RNA plugin</h2>
[ <a href="#rna_tutorial">Tutorial</a> ]<br>
[ <a href="#rna_introduction">Introduction</a> ]<br>
[ <a href="#activating_rna_mode">Activating the RNA mode</a> ]<br>
[ <a href="#rna_mode_settings">RNA mode settings</a> ]<br>
[ <a href="#rna_visualisation">RNA mode visualisation</a> ]<br>
[ <a href="#entropy">Information entropy</a> ]<br>
[ <a href="#similarity">Similarity</a> ]<br>
[ <a href="#rna_output">RNA mode file output</a> ]<br>

<a name="rna_tutorial">
	<h3>Tutorial</h3>
		If you are a first time StatAlign user and are interested in RNA secondary
		structures, the best place to start is probably the
		<a href="tutorial.pdf" target="_blank">
		StatAlign RNA tutorial</a>, which is a concise description
		of the necessary steps to take to perform a typical analysis on RNA sequences.

<a name="rna_introduction">
<h3>Introduction</h3>
<p align="justify">
The RNA plugin performs secondary structure predictions from multiple alignment samples generated by StatAlign. Both a SCFG approach (PPfold)
and a thermodynamic (RNAalifold) are available.
<br><br>The advantage of this method of secondary structure prediction is that it considers multipled sampled alignments instead of just a single fixed alignment, as is typically the
case with comparative secondary structure prediction.<br><br>

Here we describe the options available through the graphical interface.
Options available through the command line interface can be listed by running
<pre>
java -jar StatAlign.jar -help:rnaalifold
</pre>
<pre>
java -jar StatAlign.jar -help:ppfold
</pre>

</p>

The following flowchart summarises what StatAlign does when RNA mode is enabled:
<center><a href="img/flowchart1.png"><img src="img/flowchart1.png" width=400></a></center>

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="activating_rna_mode">
<h3>Activating the RNA mode</h3>
<p align="justify">
Before the RNA mode can be activated, a nucleotide alignment needs to be loaded. Once this is done the 'RNA mode' can activated by clicking on the 'RNA mode' icon below.<br>
<center>
<img src="img/nucleotide_alignment.png" width="400" height="277">
<img src="img/rna_example1_rna_mode.png" width="400" height="277"><br></center>
<br>
[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="rna_mode_settings">
<h3>RNA mode settings</h3>
<p align="justify">
Activating the RNA mode brings up a settings dialog. Two different prediction methods are available for predicting structures from multiple alignment samples. 
The first is an SCFG (Stochastic Context-Free Grammar) approach as implemented in PPfold ("Sampling and averaging (PPfold)"). 
Two predictions are produced using this approach: the standard sampling and averaging prediction, which produces a secondary structure from an averaged base-pairing probability matrix and a consensus evolutionary approach, which provides an information entropy value that takes into account contributions from alignment samples in addition to providing a secondary structure prediction.
<br><br>A thermodynamic method ("Sampling and averaging (RNAalifold)")
can also be used, this method requires that user specify the RNAalifold executable (this shouldn't be necessary on Windows). This method allows the user to specify various 
folding parameters such as the folding temperature and the genome conformation. The StatAlign GUI, however,
only provides some of these options. By running StatAlign from the command-line, additional RNAalifold parameters can be specified (see the RNAalifold manpage for a list of parameters).

<center><img src="img/rna_example2_rna_mode_settings.png" width="400" height="277"><br></center>
<br>
<br>The GUI RNAalifold options are as follows:
<br>
<ul>
	<li> <b>Temperature</b> - Rescale energy parameters to this temperature.</li>
	<li> <b>Conformation</b> - Specify whether the sequence is a linear or circular, most often the sequence is linear.</li>
	<li> <b>Covariance term</b> - Set the weight of the covariance term in the energy function.</li>
	<li> <b>Non-compatibility penalty</b> - Set the penalty for non-compatible sequences in the covariance term of the energy function.</li>
</ul>
</p>

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<h3>RNA mode run</h3>
<p align="justify">
Once the RNA mode is activated it will run alongside the normal StatAlign run.
</p>

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="rna_visualisation">
<h3>RNA mode visualisation</h3>
<p align="justify">
During and after the run there are various tabs available for secondary structure visualisation. The first is the "Consensus Structure" tab which utilises VARNA (Darty et al. (2009)) to display
the consensus secondary structure. Two modes are available: "<b>Normal</b>" mode which displays the nucleotides using a set of 4 colours to represent each base and a "<b>Probability mode</b>" which
represents unpaired nucleotides in <font color="red">red</font> and base-paired nucleotides in <font color="blue">blue</font>. Where the intensity of the <font color="red">red</font> or <font color="blue">blue</font> represents the probability that a nucleotide is unpaired or the probability that a pair of
nucleotides are base-paired, respectively.		
<center>
<img src="img/rna_example3_structure1.png" width="400" height="277">
<img src="img/rna_example3_structure2.png" width="400" height="277">
</center>	
<br><br>A second type of visualisation is available in the "Base-pairing matrix" tab which displays the average base-pairing probability matrix for alignment samples taken
up to that point in the run. The blue intensity of a cell indicates the expected probability that the particular pair of nucleotides at that position are base-paired in the structure.
<br><br>
<center><img src="img/rna_example4_matrix.png" width="400"></center>	
<br><br><b>Note:</b> only the sampling and averaging methods provide visualisations and by default only the PPfold method is shown - the RNAalifold visualisation
is shown if the PPFold method is deactivated.
</p>

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="entropy">
<h3>Information entropy</h3>
<p align="justify">
Information entropy is a measure of the spread of a probability distribution. The PPfold method is a SCFG method (Stochastic Context-Free Grammar method)
used to predict secondary structures. This SCFG method places a probability distribution on secondary structures which means that the information entropy of this 
probability distribution can be calculated  (Anderson et. al. (2012), in preparation, available upon request).
A low entropy corresponds to a probability distribution where the probability mass is concentrated on
a few structures, whereas a high entropy indicates that the probability mass is spread over many structures - this is undesirable as it can be an
indication that there are many alternative secondary structures which are almost as equiprobable as the most probable structure.
<br><br>The information entropy calculation provided in StatAlign has been extended to reflect that fact that the PPfold sampling and averaging method samples over
alignments, instead of using a single fixed alignment.
<br><br>In the 'Entropy' tab, two line graphs are displayed depicting the 'Sample Entropy', which is the information entropy of the PPfold structure predictions 
on individual alignment samples measured in bits, and 'Consensus Entropy' which is the extended information entropy that takes into account alignment space.
The 'Consensus Entropy' should approach a constant as the number of samples increases.
<br><br>
The final information entropy value can be found in the ".info" file on the header line corresponding to the consensus evolutionary prediction.
<br><br>
<center><img src="img/rna_example6_entropy.png" width="400"></center>	
</p>	

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="similarity">
<h3>Alignment similarity</h3>
<p align="justify">
In the 'Similarity' tab, a graph is shown depicting the similarity between the first alignment sample and each of the subsequent alignment samples. 
This provides a visual representation of auto-correlation between alignment samples. The similarity should gradually decrease and reach a plateau as the time between the first sample
and a given alignment becomes large, indicating that enough cycles have been taken between samples that they are no longer significantly auto-correlated.
<br><br>
<center><img src="img/rna_example7_similarity.png" width="400"></center>	
</p>			

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="rna_output">
<h3>RNA mode file output</h3>
<p align="justify">
Various secondary structure output files are available for each method in the StatAlign folder at the end of a run.
<br>The naming convention used is: <b>&#60;dataset title&#62;.&#60;method&#62;.&#60;format extension&#62;</b>. The terms are defined as follows:
<br>
<ul>
	<li> <b>&#60;dataset title&#62;</b> - the name of the input file.</li>
	<li><b>&#60;method&#62;</b> - denotes the prediction method, one of:
	<ul>
		<li><b>ppfold</b> - the PPfold sampling and averaging approach.</li>
		<li><b>cons_evol</b> - performs a structure prediction using an averaged phylogenetic probability matrix in order to calculate the information entropy value that includes the contribution from alignment samples.</li>
		<li><b>rnaalifold</b> - the RNAalifold sampling and averaging approach.</li>
		<li><b>rnaalifold_and_ppfold</b> - an averaged combination of the RNAalifold and PPfold sampling and averaging approaches.</li>
	</ul>
	<li><b>&#60;format extension&#62;</b> - one of  <b>bp</b>,  <b>dbn</b> or  <b>ct</b></li>
	<ul>
		<li><b>bp</b> - the base-pairing probability matrix as a tab-delimitted matrix file.</li>
		<li><b>dbn</b> - the maximum posterior decoding consensus structure in Vienna dot bracket format.</li>
		<li><b>ct</b> - the maximum posterior decoding consensus structure in connect format.</li>
	</ul>
</ul>
e.g. <b>'RNAData5.ppfold.ct'</b> is the maximum posterior decoding structure produced by the PPfold sampling and averaging approach for the dataset 'RNAData5' in connect format.
<br>
<br>
Finally, a summary file is also available (<b>&#60;dataset title&#62;.info</b>) that lists the maximum posterior decoding consensus structure for each method that was run,
along with some metadata fields in the title of each. The fields are described as follows:
<ul>
	<li><b>method</b> - as described above.</li>
	<li><b>title</b> - the dataset title.</li>
	<li><b>reliability_score</b> - the expected proportion of base-paired positions that are predicted to be correct.</li>
	<li><b>length</b> - the length of the structure (the number of nucleotide positions).</li>
	<li><b>entropy</b> - the information entropy of the probability distribution on structures and sampled alignments in bits.</li>
	<li><b>max_entropy</b> - the maximum possible entropy in bits.</li>
	<li><b>max_entropy</b> - the percentage of maximum entropy (entropy*100/max_entropy).</li>
</ul>
Example summary file:
<br><br>
<center><a href="#contents"><img src="img/rna_example5_info.png" width="90%" border=0></a></center>
</p>

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>
[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
<!--------------------------------------------------------------------------------------------------->



<!-- CONTENT_END -->

<script>
if(document.URL.indexOf("http:") == 0) {
document.write(' \
	</div> \
	</div> \
</div> \
');} else {
document.write('</div>');
}
</script>

</body>
</html>

