<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
    "http://www.w3.org/TR/html4/strict.dtd">
<html lang="en">
      
<head>
	<title>StatAlign &ndash; User's manual</title>

<script>
if(document.URL.indexOf("http:") == 0) {
document.write(' \
<link rel="stylesheet" type="text/css" href="../StatAlign.css"></link> \
');}
</script>

</head>
<body>

<script>
if(document.URL.indexOf("http:") == 0) {
document.write(' \
<div class="allContent"> \
	<div class="headerImage"> \
	<div class="pad"> \
		<table class="logo"> \
		<tr> \
		<td><img class="logo" alt="StatAlign" src="../StatAlignLogo.png" width="300"></td> \
		<td class="subtitle"> \
			<h3>An Extendable Software Package for Joint Bayesian Estimation of Alignments and Evolutionary Trees</h3> \
		</td> \
		</tr> \
		</table> \
	</div> \
	</div> \
 \
	<div id="menu-wrap"> \
	<div id="menu"> \
		<ul> \
			<li><a href="../index.html">Home</a></li> \
			<li><a href="../downloads.html">Downloads</a></li> \
			<li class="first"><a href="user_manual.html">User\'s manual</a></li> \
			<li><a href="developer_manual.html">For developers</a></li> \
			<li><a href="../references.html">References</a></li> \
		</ul> \
	</div> \
	</div> \
 \
	<div class="pad"> \
	<div class="container"> \
');} else {
document.write('<div style="padding: 0 5px 0 5px">');
}
</script>

<!-- CONTENT_START -->

<h2>StatAlign user's manual</h2>
<h4>For StatAlign v3.0 &ndash; last updated 3 Nov 2013</h4>

<i>The most up-to-date version of this document is available from
<a href="http://statalign.github.com/doc/user_manual.html">StatAlign's website</a>.</i>

<a name="contents">
<h2>Contents</h2>
	[ <a href="#introduction">Introduction</a> ]
		<div style='padding-left: 2em'>
		[ <a href="#tutorial">Tutorial</a> ]<br>
		[ <a href="#about_help">About this manual</a> ]<br>
		[ <a href="#about_statalign">About StatAlign</a> ]<br>
		</div>
	[ <a href="#menubar">The menu bar</a> ]<br>
		<div style='padding-left: 2em'>
		[ <a href="#file">File</a> ]<br>
		[ <a href="#analysis">Analysis</a> ]<br>
		[ <a href="#model">Model</a> ]<br>
		[ <a href="#help">Help</a> ]<br>
		</div>
	[ <a href="#settings">Analysis settings</a> ]
		<div style='padding-left: 2em'>
		[ <a href="#io_parameters">Output preferences</a> ]<br>
			<!--
			<div style='padding-left: 2em'>
			[ <a href="#logfile">Logfile</a> ]<br>
			[ <a href="#postprocess_files">Postprocess files</a> ]<br>
			[ <a href="#alignment_formats">Alignment formats</a> ]<br>
			</div>
			-->
		[ <a href="#mcmc_params">MCMC parameters</a> ]<br>
		</div>
	[ <a href="#postprocesses">The Post-processing tabulated panels</a> ]<br>
		<div style='padding-left: 2em'>
		[ <a href="#sequences">Sequences</a> ]<br>
		[ <a href="#alignment">Alignment</a> ]<br>
		[ <a href="#loglikelihood">LogLikelihood</a> ]<br>
		[ <a href="#mpd">MPD</a> ]<br>
		[ <a href="#tree">Tree</a> ]<br>
		</div>
	[ <a href="#rna_plugin">The RNA plugin</a> ]<br>
		<div style='padding-left: 2em'>
		[ <a href="#rna_introduction">Introduction</a> ]<br>
		[ <a href="#activating_rna_mode">Activating the RNA mode</a> ]<br>
		[ <a href="#rna_mode_settings">RNA mode settings</a> ]<br>
		[ <a href="#rna_visualisation">RNA mode visualisation</a> ]<br>
		[ <a href="#entropy">Information entropy</a> ]<br>
		[ <a href="#similarity">Similarity</a> ]<br>
		[ <a href="#rna_output">RNA mode file output</a> ]<br>
		</div>
	[ <a href="#example">An example</a> ]<br>

<!--------------------------------------------------------------------------------------------------->
<a name="introduction">
<h2>Introduction</h2>
	[ <a href="#tutorial">Tutorial</a> ]
	[ <a href="#about_help">About this manual</a> ]
	[ <a href="#about_statalign">About StatAlign</a> ]
	
	<a name="tutorial">
	<h3>Tutorial</h3>
		If you are a first time StatAlign user and are interested in RNA secondary
		structures, the best place to start is probably the
		<a href="https://github.com/statalign/statalign/blob/master/tutorial.pdf?raw=true" target="_blank">
		StatAlign RNA tutorial</a>, which is a concise description
		of the necessary steps to take to perform a typical sequence analysis.
		If you are working with DNA/proteins or need a more complete reference about
		StatAlign's functionality, please keep on reading.

	<a name="about_help">
	<h3>About this manual</h3>
	
		<p align="justify">
		This is a comprehensive manual of StatAlign. You can navigate 
		on this page using the hyperlinks and the <tt>Forward</tt> and
		<tt>Backward</tt> buttons of your browser. You can also access this manual
		from StatAlign in the <tt>Help</tt> &#8594; <tt>User's manual</tt> menu.
		When doing so, clicking the <tt>Home</tt> button reloads this page.<br><br>
		
		The topics are arranged in hierarchial order, the <img src="img/up.jpg" width="10" height="10" border=0>
		arrow takes you one level above.<br><br>
		
		The <img src="img/out.jpg" height=15 width=15 border=0> arrow indicates a hyperlink
		outside of the program. You cannot open such a link if your computer
		is not connected to the internet. Since these hyperlinks point to third party
		pages, we cannot guarantee that these pages will be there when you are browsing this help file.
		We also cannot guarantee that these pages will be properly loaded into our mini-browser.<br><br>
		</p>
		
		[ <a href="#introduction">Introduction</a> ] 
		<a href="#introduction"><img src="img/up.jpg" width="10" height="10" border=0></a>

	<a name="about_statalign">
	<h3>About StatAlign</h3>

		<p align="justify">
		Statistical alignment methods provide a uniform approach for inferring multiple
		sequence alignments, evolutionary trees and evolutionary parameters. Bayesian
		methods for phylogeny inferring typically fix an alignment and consider only
		substitutions as phylogenetically informative mutations. Statistical alignment
		methods	model insertions and deletions with the same background philosophy than 
		substitution models have (ie. time-continuous Markov models
		<a href="http://en.wikipedia.org/wiki/Models_of_DNA_evolution" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>		
		), but they are technically more involved. Due to these technical limitations,
		they have been becoming widespread only recently.<br><br>
	
		StatAlign provides a graphical interface for doing multiple statistical alignment
		inferring of DNA, RNA and protein sequences. It applies a Markov chain Monte Carlo
		<a href="http://en.wikipedia.org/wiki/Markov_chain_Monte_Carlo" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>
		for walking on the joint Bayesian distribution of multiple alignments, evolutionary trees
		and evolutionary parameters
		<a href="http://www.biomedcentral.com/1471-2105/9/137" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>.
		The insertion-deletion model is fixed, but the program is flexible to pair
		any implemented substitution model with the insertion-deletion model.
		The chosen insertion-deletion model is a modification of the TKF92 model
		<a href="http://www.ncbi.nlm.nih.gov/pubmed/1556741?dopt=Abstract" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>, in that it allows 'refragmentation' at internal nodes. For more details,
		see Mikl&oacute;s <i>et al.</i> (2008)
		<a href="http://www.biomedcentral.com/1471-2105/9/137" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>.
		
		<br><br>
		
		Users can load sequences, choose a substitution model with which they want to analyze
		the sequences. They also can set up the parameters of the Markov chain. Users can specify
		what kind of information they want about the samples from the MCMC chain 
		(their log-likelihood value, alignment and tree of the current state, etc.). They can
		get a consensus alignment in several formats (fasta, Clustal, Phylip and Nexus), and
		the ensemble of sampled trees in Nexus format. We suggest the sampled trees be
		further analyzed by <tt>SplitsTree4</tt>
		<a href="http://www-ab.informatik.uni-tuebingen.de/software/splitstree4/welcome.html" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>, which can read our output file.<br><br>
		
		In the next sections, we get through the graphical elements of the program (menu bar, 
		tabulated panels, pop-up dialog windows), and finally give an example how an analysis can be carried
		out.<br><br>
		</p>
		
		[ <a href="#introduction">Introduction</a> ] 
		<a href="#introduction"><img src="img/up.jpg" width="10" height="10" border=0></a>
	
	
[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
<!--------------------------------------------------------------------------------------------------->
<a name="menubar">
<h2>The menu bar</h2>

	[ <a href="#file">File</a> ]
	[ <a href="#analysis">Analysis</a> ]
	[ <a href="#model">Model</a> ]
	[ <a href="#help">Help</a> ]

	<a name="file">
	<h3>File</h3>
		
		<p align="justify">
		In the File menu, we can load sequences, set the output preferences and 
		exit the program.
		
		<h4><u>Add sequence(s)...</u></h4>
		<p align="justify">
		Clicking this menu item will result the pop up of a file opening dialog window.
		We can browse the directories in this window and open a file. The file to open 
		should contain sequences in fasta file 
		<a href="http://en.wikipedia.org/wiki/Fasta_format" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0></a>
		format. Viewing and removing the sequences can be done on the 
		<a href="#sequences"><tt>Sequences</tt></a> panel.</p>
		
		<h4><u>Output settings...</u></h4>
		<p align="justify">
		Clicking this menu item will pop up a dialog window where
		you can set up your
		<a href="#io_parameters">output preferences</a></p>
		
		<h4><u>Exit</u></h4>
		<p align="justify">
		We can leave the program here.</p>
		
		[ <a href="#menu">The menu bar</a> ] 
		<a href="#menu"><img src="img/up.jpg" width="10" height="10" border=0></a>
		<!---------------------------------------------------------------------------------->		
	<a name="analysis">
	<h3>Analysis</h3>
		
		<h4><u>Set up and run</u></h4>
		<p align="justify">
		After loading your sequences and choosing your <a href="#model">Model</a> and <a href="#io_parameters">Output settings</a>, you can click this menu item to set the MCMC parameters and start the MCMC analysis.
		The <a href="#mcmc_params">MCMC parameters</a> dialog will pop up, where you
		can set up all MCMC parameters before launching the Analysis with the
		<tt>Run!</tt> button.

		<p>Information will be given about the progress of the MCMC sampling in the Status
		bar, such as the number of burn-in steps completed and the number of MCMC samples
		taken. The <a href="#postprocesses">Postprocessing panels</a> will show both snapshots
		from the MCMC chain and also consensus entities calculated from the MCMC samples.
		These latter entities will be updated dynamically with each MCMC sample.

		<p>When the MCMC analysis terminates you will be notified with a pop-up window that
		lists all the output files created.

		<h4><u>Pause</u></h4>
		<p align="justify">
		When an MCMC is running we can suspend the running by clicking this item.
		</p>	

		<h4><u>Resume</u></h4>
		<p align="justify">
		A suspendeded run can be continued by clicking here.
		</p>	
		
		<h4><u>Stop</u></h4>
		<p align="justify">
		This item stops an MCMC run. Output files will still be created from the samples
		that have been taken so far, but you won't be able to restart the process from the
		same point. So please use it only if you indeed want to terminate the MCMC run early.
		</p>	
				
		[ <a href="#menu">The menu bar</a> ] 
		<a href="#menu"><img src="img/up.jpg" width="10" height="10" border=0></a>
		<!---------------------------------------------------------------------------------->
		
		
		<!---------------------------------------------------------------------------------->		
		
	<a name="model">
	<h3>Model</h3>

		<p align="justify">
		This menu is for selecting the substitution model that is used for analyzing the sequences.
		<i>StatAlign</i> automatically recognizes the sequences, and if sequences cannot be
		analyzed by the chosen model (for example, protein sequences are forced to be
		analyzed by a nucleotide model), it notifies the users:<br>
		<center><img src="img/bad_model.png" width=400 height=256></center><br>
		<p align="justify">
		Implemented models are:
		
		<ul>
		<li>Amino acid substitution models
			<ul> 
				<li>Blosum rate
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/8234244" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Dayhoff
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/15483331" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Chloroplast
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/10795826" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Jones-Taylor-Thornton
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/1633570" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Mammalian mitochondrial
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/9732458" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Mitochondrial reversible
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/8642615" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Retrovirus model
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/12165843" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>M&uuml;ller-Vingron model
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/11382360" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>WAG, Whelan &amp; Goldman model
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/11319253" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
			</ul>
		</li>
		<li>Nucleic acid substitution models
			<ul>		
				<li>Felsenstein81
					<a href="http://www.ncbi.nlm.nih.gov/pubmed/7288891" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Felsenstein84<br>
					Felsenstein, J. 1984. Distance methods for inferring phylogenies: a justification. Evolution 38:16Ð24.
					</li>
					
				<li>Hasegawa-Kishino-Yano  (HKY85)
					<a href="http://www.cs.manchester.ac.uk/ai/Software/phase/manual/node90.html" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Jukes-Cantor 
					<a href="http://www.geocities.com/royrajan/Web_Project/methods.html#JukesCantor" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
			
				<li>Kimura 3 parameter
					<a href="http://www.pnas.org/cgi/content/abstract/78/1/454" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>

				<li>General reversible
					<a href="http://www.cs.manchester.ac.uk/ai/Software/phase/manual/node92.html" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Tamura 1992
					<a href="http://www.cs.manchester.ac.uk/ai/Software/phase/manual/node93.html" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>
				
				<li>Tamura-Nei
					<a href="http://www.cs.manchester.ac.uk/ai/Software/phase/manual/node91.html" target="_blank">
					<img src="img/out.jpg" height=15 width=15 border=0>
					</a></li>			
			</ul>
			</li>
		</ul>
		</p>
		
		[ <a href="#menu">The menu bar</a> ] 
		<a href="#menu"><img src="img/up.jpg" width="10" height="10" border=0></a>
		<!---------------------------------------------------------------------------------->		
	<a name="help">
	<h3>Help</h3>
		The Help menu provides access to help files and other useful information in a browser
		or Java's mini-browser.
		<ul>
			<li>About...<br>
				A short summary of the authors and used libraries.</li>
			
			<li><a href="user_manual.html">User's manual</a><br>
				This document.</li>
			
			<li><a href="developer_manual.html">Developer's manual</a><br>
				A detailed description of how to implement novel plugins for StatAlign.</li>
		
			<li><a href="javadoc/index.html" target="_blank">Javadocs for StatAlign</a><br>
				Javadoc generated descriptions of the public classes, methods
				and variables of StatAlign.</li>
			
		</ul>
		
		[ <a href="#menu">The menu bar</a> ] 
		<a href="#menu"><img src="img/up.jpg" width="10" height="10" border=0></a>
		<!---------------------------------------------------------------------------------->		
	

[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
<!--------------------------------------------------------------------------------------------------->

<a name="settings">
<h2>Analysis settings</h2>

	[ <a href="#io_parameters">Output preferences</a> ]
	[ <a href="#mcmc_params">MCMC parameters</a> ]

	<a name="io_parameters">
	<h3>Output preferences</h3>

		[ <a href="#logfile">Logfile</a> ]
		[ <a href="#postprocess_files">Postprocess files</a> ]
		[ <a href="#alignment_formats">Alignment formats</a> ]

		<p align="justify">
		The output preferences are available via <tt>File</tt> &#8594; <tt>Output settings...</tt>.
		It will open this pop-up window:
		<center><img src="img/output_preferences.jpg" width=400 height=256></center><br>
		<p align="justify">	
		Below we describe the possible settings in this window.
		</p>

		<a name="logfile">
		<h4>Logfile</h4>
		
			<p align="justify">
			On the top left corner of the output preferences window, users can choose which
			postprocesses write into the logfile. The format of the output
			is the following:</p>
			<pre>
		Sample [sample_number] [TAB] postprocess_name: [TAB] data
			</pre>
			<p align="justify">
			For example:</p>
			<pre>
		Sample 0        Loglikelihood:  -3718.11540541795
		Sample 0        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.04547):0.01,(P1_1ala:0.02321,P1_1avha:0.01238):0.07713):0.01):0.03021,P1_2ran:0.08613);
		Sample 1        Loglikelihood:  -3620.5414903951587
		Sample 1        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.04547):0.01,(P1_1ala:0.02321,P1_1avha:0.07235):0.07713):0.01):0.03021,P1_2ran:0.12924);
		Sample 2        Loglikelihood:  -3618.189925496592
		Sample 2        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.04547):0.01,(P1_1ala:0.02321,P1_1avha:0.11236):0.07713):0.01):0.04352,P1_2ran:0.11424);
		Sample 3        Loglikelihood:  -3615.7331381880526
		Sample 3        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.04547):0.01,(P1_1ala:0.02321,P1_1avha:0.11236):0.07713):0.01837):0.04352,P1_2ran:0.11424);
		Sample 4        Loglikelihood:  -3587.894867267956
		Sample 4        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.06347):0.01,(P1_1ala:0.02321,P1_1avha:0.11236):0.07713):0.01837):0.04352,P1_2ran:0.11424);
		Sample 5        Loglikelihood:  -3585.9665576563943
		Sample 5        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.06347):0.01,(P1_1ala:0.02321,P1_1avha:0.11236):0.08542):0.01837):0.04352,P1_2ran:0.11902);
		Sample 6        Loglikelihood:  -3551.4723341911595
		Sample 6        Tree:   ((P1_1aeia:0.05535,((P1_1ann:0.10414,P1_1axn:0.06347):0.05301,(P1_1ala:0.02321,P1_1avha:0.11236):0.08542):0.03414):0.04352,P1_2ran:0.11902);
			.
			.
			.
			</pre>	
			</p>

		[ <a href="#io_parameters">Output preferences</a> ]
		<a href="#io_parameters"><img src="img/up.jpg" width="10" height="10" border=0></a>

		<a name="postprocess_files">
		<h4>Postprocess files</h4>

			<p align="justify">
			Users also can choose which postprocesses generate their own output file. Note
			that this file has its own format, and might be different from the data written into the log file.
			For example, the tree postrpocess writes the sampled trees into a standard Nexus file
			<a href="http://www.bioperl.org/wiki/Nexus_tree_format" target="_blank">
			<img src="img/out.jpg" height=15 width=15 border=0>
			</a>
			as its own output file.
			</p>
		[ <a href="#io_parameters">Output preferences</a> ]
		<a href="#io_parameters"><img src="img/up.jpg" width="10" height="10" border=0></a>

		<a name="alignment_formats">
		<h4>Alignment formats</h4>
		<p align="justify">
		Supported alignment formats are:
		<ul>
			<li><a name="statalign_format">StatAlign<br>
				<p align="justify">	
				It prints both the sequences at the leaves and the sequences
				predicted for the internal nodes. The sequence names and
				the alignment is separated by a TAB. The predicted sequences for the
				internal nodes contains and empty record for their name.
				Each sequence in the alignment is shown in one line. 
			</li>
			
			<li>Clustal<br>
				<p align="justify">
				The first line contains the word "Clustal".
				The alignment is broken into 60 character wide blocks.
				The sequence names are shown at the begining of the line.
				Alignment columns with all matches are marked with an
				asteriks. See also 
				<a href="http://www.bioperl.org/wiki/ClustalW_multiple_alignment_format" target="_blank">
				<img src="img/out.jpg" height=15 width=15 border=0>
				</a>		
			</li>
			
			<li>Fasta<br>
				<p align="justify">
				Alignment names are marked with a "&gt;" symbol
				and are in a separated line. Alignments are broken into
				60 character wide blocks. Each sequence are printed separately, so
				it is not easy to see which character is aligned to which one.
				See also
				<a href="http://www.bioperl.org/wiki/FASTA_multiple_alignment_format" target="_blank">
				<img src="img/out.jpg" height=15 width=15 border=0>
				</a>		
			</li>
			
			<li>Phylip<br>
				<p align="justify">
				The first line contains the number of sequences and the
				length of the alignment. The alignment is broken into 60
				character wide blocks, and a space is inserted after each 10 character long
				part. The name of the sequences are shown at the begining of the
				first block. See also
				<a href="http://www.bioperl.org/wiki/PHYLIP_multiple_alignment_format" target="_blank">
				<img src="img/out.jpg" height=15 width=15 border=0>
				</a>
			</li>
			
			<li>Nexus<br>
				<p align="justify">
				Nexus has its own <tt>XML</tt>-like format. A nexus alignment contains
				a header containing information about the seuqence type, sequence names, etc.
				See also
				<a href="http://bio.perl.org/wiki/NEXUS_multiple_alignment_format" target="_blank">
				<img src="img/out.jpg" height=15 width=15 border=0>
				</a>
			</li>
		</ul>
		</p>
		[ <a href="#io_parameters">Output preferences</a> ]
		<a href="#io_parameters"><img src="img/up.jpg" width="10" height="10" border=0></a>

	[ <a href="#settings">Analysis settings</a> ]
	<a href="#settings"><img src="img/up.jpg" width="10" height="10" border=0></a>

	<a name="mcmc_params">
	<h3>Setting and automating the MCMC parameters</h3>

		<p align="justify">
		MCMC parameters dialog window can now be opened using 
		<tt>Analysis</tt> &#8594; <tt>Set up and run</tt>.
		In this panel, the following parameters can be set before launching the MCMC run:
		<ul>
			<li>Burn-in cycles.<br>
			<p align="justify">
			The MCMC starts at a random point. This point might be far from the 
			desired distribution of alignments, trees and parameters, therefore,
			the Markov chain needs several steps to get closer the equilibrium 
			distribution. This period is called burn-in phase. Users can set
			how many steps at the begining of the Markov chain should be treated as
			the burn-in period.  
			</li>
			
			<li>Cycles after burn-in<br>
			<p align="justify">
			This is the number of Markov chain steps that the MCMC will do
			after the burn-in phase.
			</li>
			
			<li>Sampling rate<br>
			<p align="justify">
			When it is set to a number <i>c</i>, after each <i>c</i> number of
			steps during the after burn-in period, the Markov chain is sampled, 
			and the sampled states are 
			<a href="#postprocesses">postprocessed</a>. 
			</li>
			
			<li>Seed<br>
			<p align="justify">
			The Java language provides a random number generator that generates the
			same pseudo-random numbers
			<a href="http://en.wikipedia.org/wiki/Pseudorandom_number_generator" target="_blank">
			<img src="img/out.jpg" height=15 width=15 border=0>
			</a>
			starting with a so-called random seed. Users can set this number here.
			The program generates the same result when the same prameters are used,
			including the random seed generator. This is useful when one wants the
			reproduce the result of a run. When the same data should be analyzed
			with different random starting points (this is needed, for example,
			for inferring the convergence of the Markov chain), different
			seeds should be set for different runs.
			</li>
		</ul>
		
		<br><br>
		Alternatively, the user can choose to automate the process of setting one or more of the parameters by checking the corresponding checkboxes:
		<br><br>
		<center><img src="img/parameters1.png" border=0></center><br>
		Automation is not enabled by default because it may result in the termination of the MCMC process before convergence. Our recommendation is that you use the automatic settings to get a reasonable sampling setup when you don't have the resources to experiment with different MCMC settings.
		
		</p>
		
	[ <a href="#settings">Analysis settings</a> ]
	<a href="#settings"><img src="img/up.jpg" width="10" height="10" border=0></a>

[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
<!--------------------------------------------------------------------------------------------------->

<a name="postprocesses">
<h2>The Post-processing tabulated panels</h2>
	[ <a href="#sequences">Sequences</a> ]
	[ <a href="#alignment">Alignment</a> ]
	[ <a href="#loglikelihood">LogLikelihood</a> ]
	[ <a href="#mpd">MPD</a> ]
	[ <a href="#tree">Tree</a> ]

	<a name="sequences">
	<h3>Sequences</h3>

		<p align="justify">
		Sequences can be loaded from the menu, following 
		<tt>File</tt> &#8594; <tt>Add sequence(s)...</tt>.
		The loaded sequences are shown on the Sequences tabulated panel,
		in standard Fasta format. For easier visualization, the name of the
		sequences are highlighted with dark blue, and printed with different
		type of fonts.<br>
		<center><img src="img/input_sequences.png" width=400 height=256></center><br>
		<p align="justify">
		To remove a sequence from the list, click into the desired sequence and
		then to the <tt>Remove</tt> button. Adding sequences can be done simply
		by following <tt>File</tt> &#8594; <tt>Add sequence(s)...</tt>.<br>

		The sequences are automatically recognized if they are nucleotide or
		protein sequences, and a default substitution model is associated to them.
		If the sequences are not recognized, an error message is shown, indicating
		what are the unknown characters in the input sequences.</p>
	
		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>


	<a name="alignment">
	<h3>Alignment</h3>
		
		<p align="justify">
		The MCMC takes a random walk on the joint distribution of alignments, trees and
		evolutionary parameters. The current alignment can be seen on the <tt>Alignment</tt>
		panel. It is in
		<a href="#statalign_format">
		<i>
		StatAlign</i></a>
		 format.</p>
		<center><img src="img/alignment.png" width=400 height=256></center><br>

		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>

	<a name="loglikelihood">
	<h3>LogLikelihood</h3>

		<p align="justify">
		The loglikelihood trace is printed onto the screen, so the users can see the 
		convergence of the chain. The burn-in phase is colored by red, the after-burn-in
		phase is colored by blue:
		<center><img src="img/loglikelihood.jpg" width=400 height=256></center><br>
		</p>

		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>

	<a name="mpd">
	<h3>Maximum Posterior Decoding alignment</h3>
	
		<p align="justify">
		The program <i>on the fly</i> estimates Maximum Posterior Decoding
		multiple alignments
		<a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=2324098" target="_blank">
		<img src="img/out.jpg" height=15 width=15 border=0>
		</a>. This can be considered as the consensus alignment of the MCMC sampling.
		The posterior decoding values are printed on top of the alignment showing 
		certainty (reliability) of the estimated alignment column.
		<center><img src="img/mpd.png" width=400 height=256></center><br>
		<p align="justify">
		The MPD alignment is estimated using only alignment samples from the after-burn-in
		period, hence alignment is not shown during the burn-in phase in this panel.
		</p>

		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>		
		
	<a name="tree">
	<h3>Tree</h3>
	
		<p align="justify">
		This panel shows the current tree in the Markov chain.
		<center><img src="img/tree.jpg" width=400 height=256></center><br>
		</p>			

		[ <a href="#postprocesses">The Post-processing tabulated panels</a> ] 
		<a href="#postprocesses"><img src="img/up.jpg" width="10" height="10" border=0></a>


[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
<!--------------------------------------------------------------------------------------------------->

<a name="rna_plugin">
<h2>The RNA plugin</h2>
[ <a href="#rna_introduction">Introduction</a> ]
[ <a href="#activating_rna_mode">Activating the RNA mode</a> ]
[ <a href="#rna_mode_settings">RNA mode settings</a> ]
[ <a href="#rna_visualisation">RNA mode visualisation</a> ]
[ <a href="#entropy">Information entropy</a> ]
[ <a href="#similarity">Similarity</a> ]
[ <a href="#rna_output">RNA mode file output</a> ]

<a name="rna_introduction">
<h3>Introduction</h3>
<p align="justify">
The RNA plugin performs secondary structure predictions from multiple alignment samples generated by StatAlign. Both a SCFG approach (PPfold)
and a thermodynamic (RNAalifold) are available.
<br><br>The advantage of this method of secondary structure prediction is that it considers multipled sampled alignments instead of just a single fixed alignment, as is typically the
case with comparative secondary structure prediction.
</p>
The following flowchart summarises what StatAlign does when RNA mode is enabled:
<center><a href="img/flowchart1.png"><img src="img/flowchart1.png" width=400></a></center>

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="activating_rna_mode">
<h3>Activating the RNA mode</h3>
<p align="justify">
Before the RNA mode can be activated, a nucleotide alignment needs to be loaded. Once this is done the 'RNA mode' can activated by clicking on the 'RNA mode' icon below.<br>
<center>
<img src="img/nucleotide_alignment.png" width="400" height="277">
<img src="img/rna_example1_rna_mode.png" width="400" height="277"><br></center>
<br>
[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="rna_mode_settings">
<h3>RNA mode settings</h3>
<p align="justify">
Activating the RNA mode brings up a settings dialog. Two different prediction methods are available for predicting structures from multiple alignment samples. 
The first is an SCFG (Stochastic Context-Free Grammar) approach as implemented in PPfold ("Sampling and averaging (PPfold)"). 
Two predictions are produced using this approach: the standard sampling and averaging prediction, which produces a secondary structure from an averaged base-pairing probability matrix and a consensus evolutionary approach, which provides an information entropy value that takes into account contributions from alignment samples in addition to providing a secondary structure prediction.
<br><br>A thermodynamic method ("Sampling and averaging (RNAalifold)")
can also be used, this method requires that user specify the RNAalifold executable (this shouldn't be necessary on Windows). This method allows the user to specify various 
folding parameters such as the folding temperature and the genome conformation. The StatAlign GUI, however,
only provides some of these options. By running StatAlign from the command-line, additional RNAalifold parameters can be specified (see the RNAalifold manpage for a list of parameters).

<center><img src="img/rna_example2_rna_mode_settings.png" width="400" height="277"><br></center>
<br>
<br>The GUI RNAalifold options are as follows:
<br>
<ul>
	<li> <b>Temperature</b> - Rescale energy parameters to this temperature.</li>
	<li> <b>Conformation</b> - Specify whether the sequence is a linear or circular, most often the sequence is linear.</li>
	<li> <b>Covariance term</b> - Set the weight of the covariance term in the energy function.</li>
	<li> <b>Non-compatibility penalty</b> - Set the penalty for non-compatible sequences in the covariance term of the energy function.</li>
</ul>
</p>

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<h3>RNA mode run</h3>
<p align="justify">
Once the RNA mode is activated it will run alongside the normal StatAlign run.
</p>

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="rna_visualisation">
<h3>RNA mode visualisation</h3>
<p align="justify">
During and after the run there are various tabs available for secondary structure visualisation. The first is the "Consensus Structure" tab which utilizes VARNA (Darty et al. (2009)) to display
the consensus secondary structure. Two modes are available: "<b>Normal</b>" mode which displays the nucleotides using a set of 4 colours to represent each base and a "<b>Probability mode</b>" which
represents unpaired nucleotides in <font color="red">red</font> and base-paired nucleotides in <font color="blue">blue</font>. Where the intensity of the <font color="red">red</font> or <font color="blue">blue</font> represents the probability that a nucleotide is unpaired or the probability that a pair of
nucleotides are base-paired, respectively.		
<center>
<img src="img/rna_example3_structure1.png" width="400" height="277">
<img src="img/rna_example3_structure2.png" width="400" height="277">
</center>	
<br><br>A second type of visualisation is available in the "Base-pairing matrix" tab which displays the average base-pairing probability matrix for alignment samples taken
up to that point in the run. The blue intensity of a cell indicates the expected probability that the particular pair of nucleotides at that position are base-paired in the structure.
<br><br>
<center><img src="img/rna_example4_matrix.png" width="400"></center>	
<br><br><b>Note:</b> only the sampling and averaging methods provide visualisations and by default only the PPfold method is shown - the RNAalifold visualisation
is shown if the PPFold method is deactivated.
</p>

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="entropy">
<h3>Information entropy</h3>
<p align="justify">
Information entropy is a measure of the spread of a probability distribution. The PPfold method is a SCFG method (Stochastic Context-Free Grammar method)
used to predict secondary structures. This SCFG method places a probability distribution on secondary structures which means that the information entropy of this 
probability distribution can be calculated  (Anderson et. al. (2012), in preparation, available upon request).
A low entropy corresponds to a probability distribution where the probability mass is concentrated on
a few structures, whereas a high entropy indicates that the probability mass is spread over many structures - this is undesirable as it can be an
indication that there are many alternative secondary structures which are almost as equiprobable as the most probable structure.
<br><br>The information entropy calculation provided in StatAlign has been extended to reflect that fact that the PPfold sampling and averaging method samples over
alignments, instead of using a single fixed alignment.
<br><br>In the 'Entropy' tab, two line graphs are displayed depicting the 'Sample Entropy', which is the information entropy of the PPfold structure predictions 
on individual alignment samples measured in bits, and 'Consensus Entropy' which is the extended information entropy that takes into account alignment space.
The 'Consensus Entropy' should approach a constant as the number of samples increases.
<br><br>
The final information entropy value can be found in the ".info" file on the header line corresponding to the consensus evolutionary prediction.
<br><br>
<center><img src="img/rna_example6_entropy.png" width="400"></center>	
</p>	

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="similarity">
<h3>Alignment similarity</h3>
<p align="justify">
In the 'Similarity' tab, a graph is shown depicting the similarity between the first alignment sample and each of the subsequent alignment samples. 
This provides a visual representation of auto-correlation between alignment samples. The similarity should gradually decrease and reach a plateua as the time between the first sample
and a given alignment becomes large, indicating that enough cycles have been taken between samples that they are no longer significantly auto-correlated.
<br><br>
<center><img src="img/rna_example7_similarity.png" width="400"></center>	
</p>			

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>

<a name="rna_output">
<h3>RNA mode file output</h3>
<p align="justify">
Various secondary structure output files are available for each method in the StatAlign folder at the end of a run.
<br>The naming convention used is: <b>&#60;dataset title&#62;.&#60;method&#62;.&#60;format extension&#62;</b>. The terms are defined as follows:
<br>
<ul>
	<li> <b>&#60;dataset title&#62;</b> - the name of the input file.</li>
	<li><b>&#60;method&#62;</b> - denotes the prediction method, one of:
	<ul>
		<li><b>ppfold</b> - the PPfold sampling and averaging approach.</li>
		<li><b>cons_evol</b> - performs a structure prediction using an averaged phylogenetic probability matrix in order to calculate the information entropy value that includes the contribution from alignment samples.</li>
		<li><b>rnaalifold</b> - the RNAalifold sampling and averaging approach.</li>
		<li><b>rnaalifold_and_ppfold</b> - an averaged combination of the RNAalifold and PPfold sampling and averaging approaches.</li>
	</ul>
	<li><b>&#60;format extension&#62;</b> - one of  <b>bp</b>,  <b>dbn</b> or  <b>ct</b></li>
	<ul>
		<li><b>bp</b> - the base-pairing probability matrix as a tab-delimitted matrix file.</li>
		<li><b>dbn</b> - the maximum posterior decoding consensus structure in Vienna dot bracket format.</li>
		<li><b>ct</b> - the maximum posterior decoding consensus structure in connect format.</li>
	</ul>
</ul>
e.g. <b>'RNAData5.ppfold.ct'</b> is the maximum posterior decoding structure produced by the PPfold sampling and averaging approach for the dataset 'RNAData5' in connect format.
<br>
<br>
Finally, a summary file is also available (<b>&#60;dataset title&#62;.info</b>) that lists the maximum posterior decoding consensus structure for each method that was run,
along with some metadata fields in the title of each. The fields are described as follows:
<ul>
	<li><b>method</b> - as described above.</li>
	<li><b>title</b> - the dataset title.</li>
	<li><b>reliability_score</b> - the expected proportion of base-paired positions that are predicted to be correct.</li>
	<li><b>length</b> - the length of the structure (the number of nucleotide positions).</li>
	<li><b>entropy</b> - the information entropy of the probability distribution on structures and sampled alignments in bits.</li>
	<li><b>max_entropy</b> - the maximum possible entropy in bits.</li>
	<li><b>max_entropy</b> - the percentage of maximum entropy (entropy*100/max_entropy).</li>
</ul>
Example summary file:
<br><br>
<center><a href="#contents"><img src="img/rna_example5_info.png" width="90%" border=0></a></center>
</p>

[ <a href="#rna_plugin">The RNA plugin</a> ] 
<a href="#rna_plugin"><img src="img/up.jpg" width="10" height="10" border=0></a>
[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
<!--------------------------------------------------------------------------------------------------->


<a name="example">
<h2>An example</h2>
	
	<p align="justify">
	In this example, we will analyze pfcp (Papain Family Cystein Proteinase) 
	sequences. First we open the file containing 13 sequences downloaded from the
	HOMSTRAD database
	<a href="http://tardis.nibio.go.jp/homstrad/" target="_blank">
	<img src="img/out.jpg" height=15 width=15 border=0>
	</a>:
	<center><img src="img/example1.png" width=400 height=256>
	<img src="img/example2.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	We can see the loaded sequences in the <tt>Sequences</tt> panel:
	<center><img src="img/example3.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">	
	Then we set the output preferences. We select all possible posterior analyzis and
	output files, and we opt to get alignments in nexus format:
	<center><img src="img/example4.png" width=400 height=256>
	<img src="img/example5.png" width=400 height=256>
	<img src="img/example6.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	After this we start the MCMC analysis with the default parameters (10000 burn-in
	steps, 100000 steps after burn-in, sampled after each 1000 steps, random seed is set to
	1):
	<center><img src="img/example7.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	We have to give the name of the output files, and after this the analysis starts:
	<center><img src="img/example7a.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">	
	The first state is generated by an iterative alignment, which takes time. Indeed,
	this step takes as much time as a Clustal-W multiple alignment itself. <i>StatAlign</i>
	warns the users that the initialization takes time:
	<center><img src="img/example8.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">		
	We can see the loglikelihood values quickly increasing in the burn-in phase:
	<center><img src="img/example9.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	The loglikelihood trace analysis reveals that the 10000 steps were enough for convergence, 
	however, there is a significant autocorrelation, and hence 100000 steps were not enough to
	get a sufficient sample size:
	<center><img src="img/example10.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	Indeed, these sequences are very hard to analyze, as they contain several variable region.
	Small posterior probabilities on top of the MPD alignment indicate these regions:
	<center><img src="img/example11.png" width=400 height=256>
	</center><br>
	</p>
	<p align="justify">
	The sampled trees are written in nexus format, and can be opened with SplitsTree4. 
	The consensus network analysis of SplitsTree4 shows the phylogenetic uncertainty:
	<center><img src="img/example12.png" width=400 height=256>
	</center><br>
	</p>

	[ <a href="#contents">Contents</a> ] 
<a href="#contents"><img src="img/up.jpg" width="10" height="10" border=0></a>
<!--------------------------------------------------------------------------------------------------->
<br><br>

<!-- CONTENT_END -->

<script>
if(document.URL.indexOf("http:") == 0) {
document.write(' \
	</div> \
	</div> \
</div> \
');} else {
document.write('</div>');
}
</script>

</body>
</html>

